### DISCUSSION TITLE
KAMP is only working partially

### DISCUSSION THREAD

**Discussion by Heiko910** [bassamanator/everything-sovol-sv06#33] [Q&A]
Hi guys,
I have had my SV06 running with klipper for a while now. Bassamanator's GitHub was instrumental in making this happen. Thanks again.

Now I am playing around with the fine tunning and enabling of KAMP.
I have KAMP_Settings.cfg included in the "osskc.cfg", 
I have it enabled in the "misc-macros.cfg" under [gcode_macro _globals] : variable_kamp_enable: 1
I have three include lines in the "KAMP_settings.cfg" [include ./Adaptive_Meshing.cfg]  [include ./Smart_Park.cfg] [include ./Line_Purge.cfg] 

I have Label Objects enabled in Prusaslicer.
I got the "[exclude_object]" in the printer config
And I also got "enable_object_processing: True" in the Moonraker.cfg under [file_manager}

Adaptive meshing is working!
But I cannot get adaptive purging or smart parking to work.
Do I need to add anything to my slicer start Gcode?
What am I missing?
Any help would be appreciated.
Thanks

**Comment by bassamanator** [comment]
Smart parking is useful if you have a detachable probe, see official KAMP docs on how to use it.

For line purge, the command is `line_purge`, follow what I say in the `adjust your slicer` section.

**Comment by Heiko910** [comment]
Isn't the smart park also useful for heating the nozzle?
I got it to work. I just added line purge and smart park to my print start
macro.
